nixos/modules/services/x11/display-managers/sddm.nix

```diff
@@ -14,7 +14,7 @@ let
   xserverWrapper = pkgs.writeScript "xserver-wrapper" ''
     #!/bin/sh
     ${concatMapStrings (n: "export ${n}=\"${getAttr n xEnv}\"\n") (attrNames xEnv)}
-    exec ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
+    exec systemd-cat ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
```

Comment:
@groxxda I believe it was fixed in #19108, systemd-journald should just restart now on rebuild  (and restarts don't break the pipe, as was concluded in #16096).